packages/migrate/commonsChunkPlugin/commonsChunkPlugin.ts

```diff
@@ -0,0 +1,252 @@
+import {
+	addOrUpdateConfigObject,
+	createIdentifierOrLiteral,
+	createProperty,
+	findAndRemovePluginByName,
+	findPluginsByName,
+	findRootNodesByName,
+} from "@webpack-cli/utils/ast-utils";
+
+import { IJSCodeshift, INode } from "../types/NodePath";
+
+/**
+ *
+ * Transform for CommonsChunkPlugin. If found, removes the
+ * plugin and sets optimizations.namedModules to true
+ *
+ * @param {Object} j - jscodeshift top-level import
+ * @param {Node} ast - jscodeshift ast to transform
+ * @returns {Node} ast - jscodeshift ast
+ */
+export default function(j: IJSCodeshift, ast: INode): INode {
+	const splitChunksProps: INode[] = [];
+	const cacheGroupsProps: INode[] = [];
+	const optimizationProps: object = {};
+
+	let commonCacheGroupsProps: INode[] = [
+		createProperty(j, "chunks", "initial"),
+		createProperty(j, "enforce", true),
+	];
+
+	// find old options
+	const CommonsChunkPlugin: INode = findPluginsByName(j, ast, [
+		"webpack.optimize.CommonsChunkPlugin",
+	]);
+
+	if (!CommonsChunkPlugin.size()) {
+		return ast;
+	}
+
+	// cache group options based on keys
+	let cacheGroup: object = {};
+	let cacheGroups: INode[] = [];
+	let isAsyncChunk: boolean = false;
+
+	// iterate each CommonsChunkPlugin instance
+	CommonsChunkPlugin.forEach(
+		(path: INode): void => {
+			const CCPProps: INode[] = path.value.arguments[0].properties;
+
+			// reset chunks from old props
+			cacheGroup = {};
+			cacheGroups = [];
+			isAsyncChunk = false;
+			commonCacheGroupsProps = [
+				createProperty(j, "chunks", "initial"),
+				createProperty(j, "enforce", true),
+			];
+
+			let chunkKey: string;
+			let chunkCount: number = 0;
```

Comment:
If possible, I think maybe you could base this on the `idx` argument supplied as a parameter in the forEach loop somehow

Reply:
Yeah, I tried that approach before but we don't create new chunks in many cases like runtime, async, filename. We only create new chunk once when we get a name or names property.